### PR TITLE
Introduces ActiveJobSubscriber

### PR DESCRIPTION
**Pull request by hennevogel**
Implements subscriber for ActiveJob.

https://guides.rubyonrails.org/active_support_instrumentation.html#active-job

### CODE REVIEW

**Comment by hennevogel** [comment]
Need to come up with a clever way to test this. I was thinking of emitting some mocked `ActiveSupport::Notifications` on another route, like `/active_job`. What do you think?

File: lib/influxdb/rails/middleware/active_job_subscriber.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            :running
+          when "perform"
+            :succeeded
+          end
+        end
+
+        private
+
+        def values(duration, _payload)
+          value = short_hook_name == "perform.active_job" ? duration : "1"
```
Comment (ChrisBr) [inline lib/influxdb/rails/middleware/active_job_subscriber.rb]: Can we do something like `succeeded?`?

File: lib/influxdb/rails/middleware/active_job_subscriber.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            value: value,
+          }
+        end
+
+        def tags(payload)
+          {
+            hook:  short_hook_name,
+            state: job_state(payload),
+            job:   payload[:job].class.name,
+            queue: payload[:job].queue_name,
```
Comment (ChrisBr) [inline lib/influxdb/rails/middleware/active_job_subscriber.rb:21]: Can this be a whiny nil? Should we do `payload[:job]&.queue_name`?

**Review by ChrisBr — CHANGES_REQUESTED** [review]
> I was thinking of emitting some mocked ActiveSupport::Notifications on another route, like /active_job. What do you think?

Why not just creating a real active job? Would be too much overhead?

File: lib/influxdb/rails/middleware/active_job_subscriber.rb (hunk 1 above)
Comment (hennevogel) [reply lib/influxdb/rails/middleware/active_job_subscriber.rb]: You mean like 

```ruby
def measure_performance?
  short_hook_name == "perform.active_job"
end
```

File: lib/influxdb/rails/middleware/active_job_subscriber.rb (hunk 2 above)
Comment (hennevogel) [reply lib/influxdb/rails/middleware/active_job_subscriber.rb:21]: Nope can't be, I don't see how as this always comes from `ActiveJob`

**Comment by hennevogel** [comment]
> Why not just creating a real active job? Would be too much overhead?

The job is not the problem, mocking the `ActiveJob` backend is.

File: lib/influxdb/rails/middleware/active_job_subscriber.rb (hunk 2 above)
Comment (hennevogel) [reply lib/influxdb/rails/middleware/active_job_subscriber.rb:21]: Also never seen this in production...

File: lib/influxdb/rails/middleware/active_job_subscriber.rb (hunk 1 above)
Comment (hennevogel) [reply lib/influxdb/rails/middleware/active_job_subscriber.rb]: please have another look

File: lib/influxdb/rails/middleware/active_job_subscriber.rb (hunk 2 above)
Comment (ChrisBr) [reply lib/influxdb/rails/middleware/active_job_subscriber.rb:21]: Okay.

Just wondering what would happen if you delete a job class and still have some enqueued 🤔

**Review by ChrisBr — APPROVED** [review]

**Comment by ChrisBr** [comment]
@hennevogel added some specs, what do you think? The async backend does trigger enqueue, start and perform jobs so I think we're good.

**Pull request merged**